sources/packages/backend/apps/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -95,7 +101,30 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
           { entityManager: entityManager },
         );
       }
-      return this.repo.save(studentAppeal);
+      const application = await this.applicationService.getApplicationInfo(
+        applicationId,
+      );
```

Comment:
As per the lessons learned from the load test, we should not use external repo inside transaction. 

Either move this method outside the transaction, or update `getApplicationInfo` to use optional external entity manager.